conntracker panic: interface {} is nil, not *simplelru.entry

**Logs**
```
panic: interface conversion: interface {} is nil, not *simplelru.entry

goroutine 292 [running]:
github.com/hashicorp/golang-lru/simplelru.(*LRU).removeElement(0xc000dae4a0, 0xc00105b230)
	/root/go/pkg/mod/github.com/hashicorp/golang-lru@v0.5.4/simplelru/lru.go:172 +0x152
github.com/hashicorp/golang-lru/simplelru.(*LRU).removeOldest(0xc000dae4a0)
	/root/go/pkg/mod/github.com/hashicorp/golang-lru@v0.5.4/simplelru/lru.go:165 +0x4c
github.com/hashicorp/golang-lru/simplelru.(*LRU).Add(0xc000dae4a0, 0x1be7fa0, 0xc000a6db60, 0x1942260, 0xc002cd7340, 0xc000099e00)
	/root/go/pkg/mod/github.com/hashicorp/golang-lru@v0.5.4/simplelru/lru.go:67 +0x346
github.com/Kindling-project/kindling/collector/metadata/conntracker.(*conntrackCache).Add.func2(0xc004cac508, 0xc004cac4c8, 0xa6d301)
	/source/kindling/collector/metadata/conntracker/conntracker_cache.go:73 +0x15d
github.com/Kindling-project/kindling/collector/metadata/conntracker.(*conntrackCache).Add(0xc00105b200, 0xc004cac400, 0xc00089a100)
	/source/kindling/collector/metadata/conntracker/conntracker_cache.go:76 +0xe4
github.com/Kindling-project/kindling/collector/metadata/conntracker.(*Conntracker).updateCache(0xc00105b290, 0xc004cac400, 0x0)
	/source/kindling/collector/metadata/conntracker/conntracker.go:152 +0x8e
github.com/Kindling-project/kindling/collector/metadata/conntracker.(*Conntracker).poll.func2(0xc00089a120, 0xc00105b290)
	/source/kindling/collector/metadata/conntracker/conntracker.go:127 +0x51
created by github.com/Kindling-project/kindling/collector/metadata/conntracker.(*Conntracker).poll
	/source/kindling/collector/metadata/conntracker/conntracker.go:125 +0x13e
```

--- a/collector/metadata/conntracker/conntracker.go
+++ b/collector/metadata/conntracker/conntracker.go
@@ -145,23 +145,17 @@ func (ctr *Conntracker) updateCache(f *conntrack.Flow) bool {
 	if !IsNAT(f) {
 		return false
 	}
-
-	ctr.mu.Lock()
-	defer ctr.mu.Unlock()
-
 	return ctr.cache.Add(f)
 }
 
 func (ctr *Conntracker) GetDNATTuple(srcIP uint32, dstIP uint32, srcPort uint16, dstPort uint16, isUdp uint32) *IPTranslation {
-	k := connKey{
+	k := &connKey{
 		srcIP:   srcIP,
 		srcPort: srcPort,
 		dstIP:   dstIP,
 		dstPort: dstPort,
 		isUdp:   isUdp,
 	}
-	ctr.mu.RLock()
-	defer ctr.mu.RUnlock()
 
 	entry, ok := ctr.cache.Get(k)
 	if !ok {

--- a/collector/metadata/conntracker/conntracker_cache.go
+++ b/collector/metadata/conntracker/conntracker_cache.go
@@ -5,6 +5,7 @@ import (
 	"github.com/ti-mo/conntrack"
 	"golang.org/x/sys/unix"
 	"net"
+	"sync"
 	"sync/atomic"
 )
 
@@ -27,6 +28,9 @@ type IPTranslation struct {
 }
 
 type conntrackCache struct {
+	// Must use sync.Mutex instead of sync.RWMutex because
+	// cache.Add and cache.Get both modify the list
+	mu sync.Mutex
 	//TODO replace simplelru with groupcache
 	cache *simplelru.LRU
 	stats struct {
@@ -50,8 +54,10 @@ func newConntrackCache(maxStateSize int) *conntrackCache {
 	return c
 }
 
-func (cc *conntrackCache) Get(k connKey) (*IPTranslation, bool) {
+func (cc *conntrackCache) Get(k *connKey) (*IPTranslation, bool) {
 	atomic.AddInt64(&cc.stats.gets, 1)
+	cc.mu.Lock()
+	defer cc.mu.Unlock()
 	entry, ok := cc.cache.Get(k)
 	if !ok {
 		return nil, false
@@ -73,10 +79,14 @@ func (cc *conntrackCache) Add(f *conntrack.Flow) bool {
 		cc.cache.Add(key, value)
 		return true
 	}
+	cc.mu.Lock()
+	defer cc.mu.Unlock()
 	return AddToCache(&f.TupleOrig, &f.TupleReply) && AddToCache(&f.TupleReply, &f.TupleOrig)
 }
 
-func (cc *conntrackCache) Remove(k connKey) bool {
+func (cc *conntrackCache) Remove(k *connKey) bool {
+	cc.mu.Lock()
+	defer cc.mu.Unlock()
 	if cc.cache.Remove(k) {
 		atomic.AddInt64(&cc.stats.remove, 1)
 		return true
@@ -88,8 +98,8 @@ func (cc *conntrackCache) Len() int {
 	return cc.cache.Len()
 }
 
-func tupleToKey(tuple *conntrack.Tuple) (connKey, bool) {
-	k := connKey{
+func tupleToKey(tuple *conntrack.Tuple) (*connKey, bool) {
+	k := &connKey{
 		srcIP:   IPToUInt32(tuple.IP.SourceAddress),
 		dstIP:   IPToUInt32(tuple.IP.DestinationAddress),
 		srcPort: tuple.Proto.SourcePort,

